### ISSUE TITLE
Freeze after find on page jumps back to the beginning

### ISSUE THREAD

**Issue by btlechowski** [brave/browser-laptop#13732]
### Steps to Reproduce
  1. Open about:welcome
  2. Open Find on page
  3. Search 'fastest'
  4. Go forward with the search till you jump back to the beginning

**Actual result:**
Freeze. Brave takes 100% CPU then crashes without a crash dump

**Expected result:**
No Freeze

**Reproduces how often:**
100%

### Brave Version
```
Brave: 0.22.103 
V8: 6.5.254.41 
rev: 2324f5563110b178304d30b81f3cfae965ae4a03 
Muon: 6.0.0 
OS Release: 6.1.7601 
Update Channel: Beta 
OS Architecture: x64 
OS Platform: Microsoft Windows 
Node.js: 7.9.0 
Brave Sync: v1.4.2 
libchromiumcontent: 65.0.3325.181
```

### Steps to Reproduce 2
  1. Open about:welcome
  2. Open Find on page
  3. Search 'exte'
  4. Go forward with the search till you jump back to the beginning

**Comment by LaurenWags** [comment]
Using find on page on MacOS beachballed for me and I had to force quit.

**Comment by bridiver** [comment]
Fixed in muon 6.0.2

**Comment by LaurenWags** [comment]
Verified on macOS 10.12.6 x64 using the following build:
* 0.22.108 https://github.com/brave/browser-laptop/commit/2ecddeb917bfae1ecbc5ed33606924628bc299d0
* libchromiumcontent: 65.0.3325.181
* muon: 6.0.2
